// xiaoyuer/api-offer.js
var fs = require('fs');
var API = require('wechat').API;

var str = fs.readFileSync(__dirname+"/../../appConfig.json");
var appJson = JSON.parse(str);
var api = new API(appJson.AppID, appJson.AppSecret);

/*
*自定义菜单的获取
 */
exports.menu = function(callback){
    api.getMenu(callback);
}
/*
*获取二维码
* 有效时长1800s
 */
exports.qr = function(callback){
    api.createTmpQRCode(new Date().getTime(),1800,function(err,result){
        var res1 = {};
        if(!err)
            res1 = {
            ticket:result.ticket,
            code:0
            };
        callback(err,res1)
    });
}
/*
*成功生成新订单提醒需求方
 */
exports.orderSuccess = function(openid,orderID,serviceName,remark,callback){
    var templateId = 'XFVvzrEakYsnruOM6_GSTzlcGEvkNv6CXMj_0qO4X_k';
// URL置空，则在发送后,点击模板消息会进入一个空白页面（ios）, 或无法点击（android）
    var url = '';
    var topcolor = '#FF0000'; // 顶部颜色
    var data = {
        first:{
            "value":'您的订单已提交成功',
            "color":"#173177"
        },
        orderID:{
            "value":orderID,
            "color":"#173177"
        },
        orderMoneySum:{
            "value":"0.0",
            "color":"#173177"
        },
        backupFieldName:{
            "value":'服务名称',
            "color":"#173177"
        },
        backupFieldData:{
            "value":serviceName,
            "color":"#173177"
        },
        remark:{
            "value":remark ? remark:"",
            "color":"#173177"
        }
    };
    api.sendTemplate(openid, templateId, url, topcolor, data, callback);
}

/*
*当有新的订单生成时
* 提醒服务方
 */
exports.neworder = function(openid,customInfo,serviceName,remark,callback){
    var templateId = 'xJ5GAvrvDdCUpzbSYW8YxyyKNeFzlu3KmInvwH0XynY';
// URL置空，则在发送后,点击模板消息会进入一个空白页面（ios）, 或无法点击（android）
    var url = '';
    var topcolor = '#FF0000'; // 顶部颜色
    var data = {
        first:{
            "value":'您收到了一条新的订单',
            "color":"#173177"
        },
        tradeDateTime:{
            "value":new Date().Format("yy-MM-dd hh:mm:ss"),
            "color":"#173177"
        },
        customerInfo:{
            "value":customInfo,
            "color":"#173177"
        },
        orderItemName:{
            "value":'服务名称',
            "color":"#173177"
        },
        orderItemData:{
            "value":serviceName,
            "color":"#173177"
        },
        remark:{
            "value":remark ? remark:"",
            "color":"#173177"
        }
    };
    api.sendTemplate(openid, templateId, url, topcolor, data, callback);
}
/*
*用户支付时给出提示
 */
exports.pay = function(openid,tradeType,remark,curAmount,callback){
    var templateId = 'OkLhAIffCzN5PdCo1R-Zmm49kEFx0rHb3hercBv7lMM';
// URL置空，则在发送后,点击模板消息会进入一个空白页面（ios）, 或无法点击（android）
    var url = '';
    var topcolor = '#FF0000'; // 顶部颜色
    var data = {
        first:{
            "value":'实时交易提醒',
            "color":"#173177"
        },
        tradeDateTime:{
            "value":new Date().Format("yy-MM-dd hh:mm:ss"),
            "color":"#173177"
        },
        tradeType:{
            "value":tradeType,
            "color":"#173177"
        },
        curAmount:{
            "value":curAmount,
            "color":"#173177"
        },
        remark:{
            "value":remark ? remark:"",
            "color":"#173177"
        }
    };
    api.sendTemplate(openid, templateId, url, topcolor, data, callback);
}

/*
*确认订单通知
 */
exports.ensure  =function(openid,orderId,remark,callback){
    var templateId = '55xw4iYQy_l60kQjrTKCv0ysk4TED7A1ck2Qmcgo4j8';
// URL置空，则在发送后,点击模板消息会进入一个空白页面（ios）, 或无法点击（android）
    var url = '';
    var topcolor = '#FF0000'; // 顶部颜色
    var data = {
        first:{
            "value":'订单状态变更通知',
            "color":"#173177"
        },
        OrderSn:{
            "value":orderId,
            "color":"#173177"
        },
        OrderStatus:{
            "value":"已确认",
            "color":"#173177"
        },
        remark:{
            "value":remark ? remark:"",
            "color":"#173177"
        }
    };
    api.sendTemplate(openid, templateId, url, topcolor, data, callback);
}
Date.prototype.Format = function (fmt) { //author: meizz
    var o = {
        "M+": this.getMonth() + 1, //月份
        "d+": this.getDate(), //日
        "h+": this.getHours(), //小时
        "m+": this.getMinutes(), //分
        "s+": this.getSeconds(), //秒
        "q+": Math.floor((this.getMonth() + 3) / 3), //季度
        "S": this.getMilliseconds() //毫秒
    };
    if (/(y+)/.test(fmt)) fmt = fmt.replace(RegExp.$1, (this.getFullYear() + "").substr(4 - RegExp.$1.length));
    for (var k in o)
        if (new RegExp("(" + k + ")").test(fmt)) fmt = fmt.replace(RegExp.$1, (RegExp.$1.length == 1) ? (o[k]) : (("00" + o[k]).substr(("" + o[k]).length)));
    return fmt;
};
