// xiaoyuer/xiaoyuer/seek.js

var request = require('request');

/*
*按照类型获取服务列表
*
 */
function seekService(type,callback){
    var url = baseUrl + "/service/list?type=" + type;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}

/*
*根据id获取服务信息
 */
function serviceInfo(id,callback){
    var url = baseUrl + "/service/info?id=" + id;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}

/*
*根据id和openid收藏服务
 */
function collectService(id,openid,callback){
    var url = baseUrl + "/service/collect?id=" + id + "&openid=" + openid;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}

function seekGame(type,callback){
    var url = baseUrl + "/game/list?type=" + type;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}

function gameInfo(id,callback){
    var url = baseUrl + "/game/info?id=" + id;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}

function collectGame(id,openid,callback){
    var url = baseUrl + "/game/collect?id=" + id + "&openid=" + openid;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}

function seekCommonweal(type,callback){
    var url = baseUrl + "/commonweal/list?type=" + type;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}

function commonwealInfo(id,callback){
    var url = baseUrl + "/commonweal/info?id=" + id;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}


function collectCommonweal(id,openid,callback){
    var url = baseUrl + "/commonweal/collect?id=" + id + "&openid=" + openid;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}

function seekNeed(type,callback){
    var url = baseUrl + "/need/list?type=" + type;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}

function needInfo(id,callback){
    var url = baseUrl + "/need/info?id=" + id;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}


function collectNeed(id,openid,callback){
    var url = baseUrl + "/need/collect?id=" + id + "&openid=" + openid;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error){

            }
            else
                callback(resultObj);
        }
    })
}

exports.collectNeed = collectNeed;
exports.needInfo = needInfo;
exports.seekNeed = seekNeed;
exports.collectCommonweal = collectCommonweal;
exports.commonwealInfo = commonwealInfo;
exports.seekNCommonweal = seekCommonweal;
exports.collectGame = collectGame;
exports.gameInfo = gameInfo;
exports.seekGame = seekGame;
exports.collectService = collectService;
exports.serviceInfo = serviceInfo;
exports.seekService = seekService;
