// xiaoyuer/xiaoyuer/userInfo.js
var request = require("request");

/*
*获取用户基本信息
 */
function getUserInfo(openid,callback){
    var url = baseUrl + "/user/info?openid=" + openid;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error)
                {}
            else
              callback(resultObj);
            /*
            若已绑定返回信息{code:"0"}，否则返回{"code":-1}
             */
        }
    })
}

/*
*获取用户余额
 */
function getBalance(openid,callback){
    var url = baseUrl + "/user/balance?openid=" + openid;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error)
            {}
            else
                callback(resultObj.balance);
        }
    })
}
/*
*获取订单列表
 * 分页获取
 */
function getOrder(openid,page,callback){
    var url = baseUrl + "/user/orders?openid=" + openid + "&page=" + page;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error)
            {}
            else
                callback(resultObj);
        }
    })
}

/*
*用户扫描二维码之后的处理
* 不管是否关注，均传ticket和openid
 */
function scanQR(openid,ticket,callback){
    var url = baseUrl + "/user/qr?openid=" + openid + "&ticket=" + ticket;
    /*
       若成功则返回{"code":1}
     */
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error)
                callback(0);
            else
                callback(resultObj);
        }
    })
}
/*
*取消openid与小鱼儿账户的绑定
 */
function cancelBind(openid,callback){
    var url = baseUrl + "/user/cancel?openid=" + openid ;
    request(url,function(error,response,body){
        if (!error && response.statusCode==200){
            var resultObj = JSON.parse(body);
            if(body.error)
            {}
            else
                callback(resultObj);
        }
    })
}
exports.getUserInfo = getUserInfo;
exports.getBalance = getBalance;
exports.getOrder = getOrder;
exports.scanQR = scanQR;
exports.cancelBind = cancelBind;
